### PR TITLE
integ-tests: test ebs volume created from snapshot with larger size

### PR DESCRIPTION
**Pull request by chenwany**
*Issue #, if available:*

*Description of changes:*

-  Add integration test to test the size of volume matches its actual size after it is created from a snapshot.
- Change the error message of ebs_volume_size_snapshot_validator when the volume size is smaller than the snapshot size. Give a more detailed and clear error message by mentioning the section label, snapshot size and snapshot id in the error message.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 1]
```diff
@@ -131,6 +132,15 @@ def _test_home_correctly_shared(remote_command_executor, scheduler_commands):
     verify_directory_correctly_shared(remote_command_executor, "/home", scheduler_commands)
 
 
+def _test_ebs_resize(remote_command_executor, mount_dir, volume_size):
+    logging.info("Testing ebs has correct volume size")
+    result = remote_command_executor.run_remote_command(
+        "VOLUME=$(lsblk -no pkname `df -h -t ext4 | tail -n +2 |grep '{size}' | awk '{{print $1}}'`)"
+        "&&lsblk | tail -n +2 | grep $VOLUME| awk '{{print $4}}' | sed -n '1p'''".format(size=mount_dir)
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_ebs.py:139]: This is a lot of commands piped together. Can this be broken down further?

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def test_ebs_snapshot(
 ):
     logging.info("Testing ebs snapshot")
     mount_dir = "ebs_mount_dir"
-    volume_size = 10
+    volume_size = 21
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_ebs.py:48]: Can you add a comment above this line explaining why you chose this value?

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 3]
```diff
@@ -131,6 +132,42 @@ def _test_home_correctly_shared(remote_command_executor, scheduler_commands):
...
+    #     Filesystem      Size  Used Avail Use% Mounted on
+    # /dev/nvme1n1p1  9.8G   37M  9.3G   1% /ebs_mount_dir
+    # /dev/nvme2n1p1  9.8G   37M  9.3G   1% /existing_mount_dir
+
+    # get the volume name given the filesystem name
+    volume_name = remote_command_executor.run_remote_command(
+        "lsblk -no pkname {filesystem_name}".format(filesystem_name=filesystem_name)
+    ).stdout
+    # example input: /dev/nvme1n1p1
+    # example output: nvme1n1
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_ebs.py:152]: same here: can you move the example output above the code that runs the command?

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 4]
```diff
@@ -131,6 +132,42 @@ def _test_home_correctly_shared(remote_command_executor, scheduler_commands):
...
+    )
+    # example output of "lsblk"
+    # NAME          MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
+    # nvme0n1       259:0    0  25G  0 disk
+    # ├─nvme0n1p1   259:1    0  25G  0 part /
+    # └─nvme0n1p128 259:2    0   1M  0 part
+    # nvme1n1       259:3    0  21G  0 disk
+    # └─nvme1n1p1   259:4    0  10G  0 part /ebs_mount_dir
+    # nvme2n1       259:5    0  10G  0 disk
+    # └─nvme2n1p1   259:6    0  10G  0 part /existing_mount_dir
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_ebs.py:166]: same here: can you move the example output above the code that runs the command?

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 5]
```diff
@@ -131,6 +132,42 @@ def _test_home_correctly_shared(remote_command_executor, scheduler_commands):
     verify_directory_correctly_shared(remote_command_executor, "/home", scheduler_commands)
 
 
+def _test_ebs_resize(remote_command_executor, mount_dir, volume_size):
+    logging.info("Testing ebs has correct volume size")
```
[the remark this answers is not in the document]
Comment (tilne) [reply tests/integration-tests/tests/storage/test_ebs.py:136]: nevermind, sorry

File: tests/integration-tests/tests/storage/test_ebs.py (hunk 2 above)
Comment (chenwany) [reply tests/integration-tests/tests/storage/test_ebs.py:48]: Yes, done!

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 6]
```diff
@@ -131,6 +132,42 @@ def _test_home_correctly_shared(remote_command_executor, scheduler_commands):
...
+
+    # get the filesystem that the shared_dir is mounted on
+    filesystem_name = remote_command_executor.run_remote_command(
+        "df -h -t ext4 | tail -n +2 |grep '{mount_dir}' | awk '{{print $1}}'".format(mount_dir=mount_dir)
+    ).stdout
+    # example output of "df -h -t ext4"
+    #     Filesystem      Size  Used Avail Use% Mounted on
+    # /dev/nvme1n1p1  9.8G   37M  9.3G   1% /ebs_mount_dir
+    # /dev/nvme2n1p1  9.8G   37M  9.3G   1% /existing_mount_dir
```
[the remark this answers is not in the document]
Comment (chenwany) [reply tests/integration-tests/tests/storage/test_ebs.py:148]: done!

File: tests/integration-tests/tests/storage/test_ebs.py (hunk 4 above)
Comment (chenwany) [reply tests/integration-tests/tests/storage/test_ebs.py:169]: done!

**Review by tilne — APPROVED** [review]
LGTM :+1:

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 7]
```diff
@@ -45,7 +45,8 @@ def test_ebs_snapshot(
 ):
     logging.info("Testing ebs snapshot")
     mount_dir = "ebs_mount_dir"
-    volume_size = 10
+    volume_size = 21
+    # This volume_size is set to be larger than snapshot size(10G), to test create volumes larger than its snapshot size
```
[the remark this answers is not in the document]
Comment (chenwany) [reply tests/integration-tests/tests/storage/test_ebs.py:49]: 👍 oh I see! I would pay attention to that next time

**Pull request merged**